File: doc/codes/python/examples/api/command/pattern.py
from autd3.commands import ChangePatternBank, ConfigPattern, Pattern, WritePatternBuffer
from autd3.geometry import Autd3, Geometry
from autd3.value import Intensity, LoopBehavior, PatternBank, SamplingConfig, TransitionMode

geometry = Geometry([Autd3([0.0, 0.0, 0.0], [1.0, 0.0, 0.0, 0.0])])

bank = PatternBank.B0
transition_mode = TransitionMode.Immediate

phases = geometry.phase_buffer()
intensities = Intensity.MAX

# ANCHOR: api
Pattern(phases, intensities)

Pattern(phases, intensities, bank=bank)

Pattern(
    phases,
    intensities,
    bank=bank,
    transition_mode=transition_mode,
)
# ANCHOR_END: api

# ANCHOR: equivalent
WritePatternBuffer(
    bank=bank,
    index=0,
    phases=phases,
    intensities=intensities,
)
ConfigPattern(
    bank=bank,
    config=SamplingConfig(0xFFFF),
    size=1,
    loop_behavior=LoopBehavior.Infinite,
)
ChangePatternBank(
    bank=bank,
    transition_mode=transition_mode,
)
# ANCHOR_END: equivalent

File: doc/codes/csharp/examples/api/command/pattern.cs
using System.Numerics;
using AUTD3;

namespace DocSamples.ApiCommandPattern;

internal static class Sample
{
    internal static void Run()
    {
        var geometry = new Geometry(new[] { new Autd3(Vector3.Zero) });

        var bank = PatternBank.B0;
        var transitionMode = TransitionMode.Immediate;

        var phases = geometry.PhaseBuffer();
        var intensities = Intensity.Max;

        // ANCHOR: api
        new Pattern(phases, intensities);

        new Pattern(bank, phases, intensities);

        new Pattern(
            bank: bank,
            phases: phases,
            intensities: intensities,
            transitionMode: transitionMode
        );
        // ANCHOR_END: api

        // ANCHOR: equivalent
        new WritePatternBuffer(
            bank: bank,
            index: 0,
            phases: phases,
            intensities: intensities
        );
        new ConfigPattern(
            bank: bank,
            config: new SamplingConfig(ushort.MaxValue),
            size: 1,
            loopBehavior: LoopBehavior.Infinite
        );
        new ChangePatternBank(
            bank: bank,
            transitionMode: transitionMode
        );
        // ANCHOR_END: equivalent
    }
}
